doc/releases/changelog-0.38.0.md

```diff
@@ -4,112 +4,362 @@
+  but the values are `qml.wires.Wires` instances.
 
-* A new function `qml.registers` has been added, enabling the creation of registers, which are implemented as a dictionary of `Wires` instances.
-  [(#5957)](https://github.com/PennyLaneAI/pennylane/pull/5957)
-  [(#6102)](https://github.com/PennyLaneAI/pennylane/pull/6102)
+  Nesting registers within other registers can be done by providing a nested dictionary, where the ordering 
+  of wire labels is based on the order of appearance and nestedness.
+
+  ```python
+  >>> wire_reg = qml.registers({"alice": {"alice1": 1, "alice2": 2}, "bob": {"bob1": 2, "bob2": 1}})
+  >>> wire_reg
+  {'alice1': Wires([0]), 'alice2': Wires([1, 2]), 'alice': Wires([0, 1, 2]), 'bob1': Wires([3, 4]), 'bob2': Wires([5]), 'bob': Wires([3, 4, 5])}
+  ```
+
+  Since the values of the dictionary are `Wires` instances, their use within quantum circuits is very 
+  similar to that of a `list` of integers.
+
+  ```python
+  dev = qml.device("default.qubit")
+
+  @qml.qnode(dev)
+  def circuit():
+      for w in wire_reg["alice"]:
+          qml.Hadamard(w)
+
+      for w in wire_reg["bob1"]:
+          qml.RX(0.1967, wires=w)
+
+      qml.CNOT(wires=[wire_reg["alice1"][0], wire_reg["bob2"][0]])
+
+      return [qml.expval(qml.Y(w)) for w in wire_reg["bob1"]]
+
+  print(qml.draw(circuit)())
+  ```
+
+  ```pycon
+  0: ──H────────╭●─┤     
+  1: ──H────────│──┤     
+  2: ──H────────│──┤     
+  3: ──RX(0.20)─│──┤  <Y>
+  4: ──RX(0.20)─│──┤  <Y>
+  5: ───────────╰X─┤  
+  ```
+
+  In tandem with `qml.registers`, we've also made the following improvements to `qml.wires.Wires`:
+
+  * `Wires` instances now have a more copy-paste friendly representation when printed.
+    [(#5958)](https://github.com/PennyLaneAI/pennylane/pull/5958)
+
+    ```python
+    >>> from pennylane.wires import Wires
+    >>> w = Wires([1, 2, 3])
+    >>> w
+    Wires([1, 2, 3])
+    ```
+
+  * Python set-based combinations are now supported by `Wires`.
+    [(#5983)](https://github.com/PennyLaneAI/pennylane/pull/5983)
+
+    This new feature unlocks the ability to combine `Wires` instances in the following ways:
+
+    * intersection with `&` or `intersection()`:
+
+      ```python
+      >>> wires1 = Wires([1, 2, 3])
+      >>> wires2 = Wires([2, 3, 4])
+      >>> wires1.intersection(wires2) # or wires1 & wires2
+      Wires([2, 3])
+      ```
+
+    * symmetric difference with `^` or `symmetric_difference()`:
+
+      ```python
+      >>> wires1.symmetric_difference(wires2) # or wires1 ^ wires2
+      Wires([1, 4])
+      ```
+
+    * union with `|` or `union()`:
+
+      ```python
+      >>> wires1.union(wires2) # or wires1 | wires2
+      Wires([1, 2, 3, 4])
+      ```
+
+    * difference with `-` or `difference()`:
+
+      ```python
+      >>> wires1.difference(wires2) # or wires1 - wires2
+      Wires([1])
+      ```
 
 <h4>Quantum arithmetic operations 🧮</h4>
```

Comment:
@josh146 @DSGuala thoughts on how this is restructured?

Reply:
Much easier to follow this way :+1: